Add config flag and support .yaml #29

Mctalian opened the pull request:
Should handle #29 if you want to use `.ls-lint.yaml` instead. Also allows for specifying a config file `ls-lint -config ./relative/path/to/config/.ls-lint.yaml`.

You can specify the relative path to a new working directory with `-pwd` flag. This could be useful for multi-language monorepos where projects have different directory naming conventions. `ls-lint -pwd ./relative/path -config ./relative/path/.ls-lint.yml`

Hyperfine performance results (2022/10/07 1:30 PM MT)

| build | results |
| ----- | ------ |
| master | `Time (mean ± σ):      85.0 ms ±   1.8 ms    [User: 48.4 ms, System: 32.4 ms]` | 
| this branch | `Time (mean ± σ):      84.6 ms ±   2.1 ms    [User: 48.4 ms, System: 32.2 ms]` |

Run on:
```
Hardware Overview:

      Model Name: MacBook Pro
      Model Identifier: MacBookPro16,1
      Processor Name: 6-Core Intel Core i7
      Processor Speed: 2.6 GHz
      Number of Processors: 1
      Total Number of Cores: 6
      L2 Cache: 768 KB
      L3 Cache: 6 MB
      Hyper-Threading Technology: Enabled
      Memory: 16 GB
```

I didn't see any contribution guidelines, so let me know if you'd like me to do something else.

loeffel-io commented:
thanks @Mctalian for the PR! To specify a config file should not change the root - maybe there should be another option for this? 🤔

Mctalian commented:
> 



> thanks @Mctalian for the PR! To specify a config file should not change the root - maybe there should be another option for this? 🤔

Sure! I can look into changing that.

Any tips on how to get past the codecov checks? I can't test the `main` func so I'm a bit stuck 😅

Mctalian commented:
@loeffel-io I added the `-pwd` flag which will change the root. `-config` no longer changes the root.

I moved the logging of errors out of main to try to increase coverage - it did increase, but not enough to pass the patch check. Not sure how much more I can increase that coverage 🙁

Mctalian commented:
Bump @loeffel-io

groenroos commented:
Keen to see this merged and released, as being able to define a custom config file path would unblock us! 🙏

Mctalian commented:
Bump @loeffel-io

Mctalian commented:
Hey @loeffel-io anything that I can do to get this merged in?

Mctalian commented:
@loeffel-io bump

loeffel-io commented:
the complexity of this pr in case of maintenance is to high for this two simple flags

Mctalian commented:
Super disappointed with that decision. Best of luck on maintaining this library. 👋🏼

loeffel-io commented:
With `v2.x` this is provided by the `config` and `workdir` option: https://github.com/loeffel-io/ls-lint/releases/tag/v2.0.0-beta.0

Feel free to try it out
